### PR TITLE
NXP-26978: cleanup code and enable linting and code formatting

### PR DESCRIPTION
**Pull request by Gabez0r**

### CODE REVIEW

File: addons/nuxeo-dam/search/assets/nuxeo-assets-search-results.html [hunk 1]
```diff
@@ -128,13 +147,12 @@
...
       properties: {
-        nxProvider: Object
+        nxProvider: Object,
       },
 
-      _navigate: function(e) {
-        this.fire('navigate', {doc: (e.model || e.detail).item});
-      }
+      _navigate(e) {
+        this.fire('navigate', { doc: (e.model || e.detail).item });
```
Comment (nelsonsilva) [inline addons/nuxeo-dam/search/assets/nuxeo-assets-search-results.html]: not a big fan of this extra space here, normally only use when in destructuring

File: addons/nuxeo-drive/elements/nuxeo-drive-edit-button.js [hunk 2]
```diff
@@ -58,49 +59,52 @@ Polymer({
...
     return [
       'nxdrive://edit',
       baseUrl.replace('://', '/'), // XXX replaceFirst
-      'user', this.user.id,
-      'repo', this.document.repository,
-      'nxdocid', this.document.uid,
-      'filename', encodeURIComponent(this.blob.name),
-      'downloadUrl', downloadUrl].join('/');
-  }
+      'user',
```
Comment (nelsonsilva) [inline addons/nuxeo-drive/elements/nuxeo-drive-edit-button.js]: :bleedingeyes:

File: addons/amazon-s3-online-storage/index.js [hunk 3]
```diff
@@ -154,12 +157,11 @@ class S3Provider {
 
   batchExecute(operationId, params, headers) {
     return this.connection.operation(operationId).then((operation) => {
-      var options = {
-        url: [operation._nuxeo._restURL, 'upload', this.batchId, 'execute', operationId].join('/')
-            .replace(/\/+/g, '\/')
+      const options = {
+        url: [operation._nuxeo._restURL, 'upload', this.batchId, 'execute', operationId].join('/').replace(/\/+/g, '/'),
```
[the remark this answers is not in the document]
Comment (Gabez0r) [reply addons/amazon-s3-online-storage/index.js]: It fits the 120 chars limit 🤷‍♂️

File: addons/nuxeo-ai-core/index.js [hunk 4]
```diff
@@ -30,6 +31,17 @@ const AISuggestionManager = (() => {
   let _updateDebouncer = null;
   let op = null;
 
+  function _getSuggestionWidget(element) {
```
[the remark this answers is not in the document]
Comment (Gabez0r) [reply addons/nuxeo-ai-core/index.js]: yes

File: addons/nuxeo-dam/search/assets/nuxeo-assets-search-results.html (hunk 1 above)
Comment (Gabez0r) [reply addons/nuxeo-dam/search/assets/nuxeo-assets-search-results.html]: We've been using it on elements for a long time.

File: addons/easyshare/document/easysharefolder/nuxeo-easysharefolder-create-layout.html [hunk 5]
```diff
@@ -22,31 +22,44 @@
...
-    <nuxeo-input role="widget" name="email" type="email"
-                 label="[[i18n('easysharefolder.contactEmail')]]"
-                 value="{{document.properties.eshare:contactEmail}}">
+    <nuxeo-input
+      role="widget"
+      name="email"
+      type="email"
+      label="[[i18n('easysharefolder.contactEmail')]]"
+      value="{{document.properties.eshare:contactEmail}}"
+    >
```
[the remark this answers is not in the document]
Comment (Gabez0r) [reply addons/easyshare/document/easysharefolder/nuxeo-easysharefolder-create-layout.html]: agree T_T

File: addons/amazon-s3-online-storage/index.js [hunk 6]
```diff
@@ -104,34 +105,36 @@ class S3Provider {
...
+    return UploaderBehavior.getProviders().default.prototype.accepts;
   }
 
   upload(files, callback) {
-    this._ensureBatch().then(() => {
-      if ((new Date()).getTime() >= this.extraInfo.expiration) {
-        this._refreshBatchInfo();
-      }
+    this._ensureBatch()
+      .then(() => {
```
[the remark this answers is not in the document]
Comment (Gabez0r) [reply addons/amazon-s3-online-storage/index.js]: yes

**Pull request merged**